src/Microsoft.AspNetCore.Server.Kestrel/Infrastructure/Constants.cs

```diff
@@ -1,14 +1,33 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System;
+using Microsoft.Extensions.PlatformAbstractions;
+
 namespace Microsoft.AspNetCore.Server.Kestrel.Infrastructure
 {
     internal class Constants
     {
         public const int ListenBacklog = 128;
 
         public const int EOF = -4095;
-        public const int ECONNRESET = -4077;
+        public static int ECONNRESET
+        {
+            get
+            {
+                switch (PlatformServices.Default.Runtime.OperatingSystemPlatform)
+                {
+                    case Platform.Windows:
+                        return -4077;
+                    case Platform.Linux:
+                        return -104;
+                    case Platform.Darwin:
+                        return -54;
+                    default:
+                        throw new Exception("Unable to determine ECONNRESET value on this platform.");
```

Comment:
Is it wise to throw here? There is absolutely nothing that a user can do about it.